### PR TITLE
Test suite improvements

### PR DESCRIPTION
**Pull request by dorileo**
+ update valgrind supressions
+ make the test runner more flexible
+ split github actions executions

### CODE REVIEW

File: scripts/run-test-suite.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	    if [ "$fullConfigSet" = false ]; then
+		configs+=($systemdConfig)
+	    fi;;
+	"shim-systemd-boot")
+	    if [ "$fullConfigSet" = false ]; then
+		configs+=($shimSystemdConfig)
+	    fi;;
+	"all")
+	    set_full_config;;
+	"--no-coverage")
```
Comment (phmccarty) [inline scripts/run-test-suite.sh]: This flag should be documented in `print_help`.

File: scripts/run-test-suite.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+set_full_config() {
+    fullConfigSet=true
+    configs=("" $systemdConfig $shimSystemdConfig)
+}
+
+for curr in "$@"; do
+    case $curr in
+	"stock")
+	    if [ "$fullConfigSet" = false ]; then
```
Comment (phmccarty) [inline scripts/run-test-suite.sh]: For consistency with the original version of the script, I would use `[[ ... ]]` instead. The same comment applies to a few other lines of the script.

File: scripts/run-test-suite.sh (hunk 1 above)
Comment (dorileo) [reply scripts/run-test-suite.sh]: sure, thank you.

File: scripts/run-test-suite.sh (hunk 2 above)
Comment (dorileo) [reply scripts/run-test-suite.sh]: done. thanks.

**Comment by dorileo** [comment]
@phmccarty fixed the issues, merging. Thanks.

**Pull request merged**